include/deal.II/distributed/cell_weights.h

```diff
@@ -0,0 +1,170 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_distributed_cell_weights_h
+#define dealii_distributed_cell_weights_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/hp/dof_handler.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace parallel
+{
+  namespace distributed
+  {
+    /**
+     * Anytime a parallel::distributed::Triangulation is repartitioned, either
+     * upon request or by refinement/coarsening, cells will be distributed
+     * amongst all subdomains to achieve an equally balanced workload. If the
+     * workload per cell varies, which is in general the case for hp::DoFHandler
+     * objects, we can take those into account by introducing individual weights
+     * for different cells.
+     *
+     * This class allows to use information about the FiniteElement on each cell
+     * to determine the weight through the hp::DoFHandler to the connected
+     * parallel::distributed::Triangulation, in which callback function of this
```

Comment:
this sentence has a complicated grammar and I don't understand it :-) Can you rewrite it?